packages/launch/src/focused-launch/domain-details/index.tsx

```diff
@@ -3,24 +3,92 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import * as React from 'react';
 import { Link } from 'react-router-dom';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
+import DomainPicker, { ITEM_TYPE_BUTTON } from '@automattic/domain-picker';
+import type { DomainSuggestions } from '@automattic/data-stores';
+import { Title, SubTitle } from '@automattic/onboarding';
+import { recordTracksEvent } from '@automattic/calypso-analytics';
+import { Icon, chevronLeft } from '@wordpress/icons';
 
 /**
  * Internal dependencies
  */
 import { Route } from '../route';
+import { useSite, useDomainSearch } from '../../hooks';
+import { LAUNCH_STORE } from '../../stores';
+import { FOCUSED_LAUNCH_FLOW_ID } from '../../constants';
 
 import './style.scss';
 
-const DomainDetails: React.FunctionComponent = () => {
+const ANALYTICS_UI_LOCATION = 'domain_step';
+
+const DomainStep: React.FunctionComponent = () => {
+	const { plan, domain } = useSelect( ( select ) => select( LAUNCH_STORE ).getState() );
+	const { currentDomainName } = useSite();
+	const domainSearch = useDomainSearch();
+
+	const {
+		setDomain,
+		unsetDomain,
+		setDomainSearch,
+		unsetPlan,
+		confirmDomainSelection,
+	} = useDispatch( LAUNCH_STORE );
```

Comment:
We may want to "abstract" the `select`/`dispatch` calls to the `LAUNCH_STORE` in a hook, similarly to what we do with `useSite()`. This would allow us to refactor some code in the Summary view too.

Although I know we also have issue #47233 — @razvanpapadopol , do you think the refactor to a `useLaunch()` hook should happen in this PR, or when working on #47233?

Reply:
>We may want to "abstract" the select/dispatch calls to the LAUNCH_STORE in a hook, similarly to what we do with useSite(). This would allow us to refactor some code in the Summary view too.

As per our brief Slack discussion, we are in agreement that we should do this within this PR 👍🏻

Reply:
> We may want to "abstract" the select/dispatch calls to the LAUNCH_STORE in a hook, similarly to what we do with useSite(). This would allow us to refactor some code in the Summary view too.
>
> Although I know we also have issue #47233 — @razvanpapadopol , do you think the refactor to a useLaunch() hook should happen in this PR, or when working on #47233?

I strongly encourage abstracting bundles of multiple selectors and/or additional logic to a hook that is reusable between flows (focused and step-by-step launch in this case) and the perfect example would be `use-domain-search`. 

But at the same time, there is no real benefit from extracting just a select/dispatch call if there is no extra processing that we could include in the abstraction.